### PR TITLE
Classification heads extension by optional linear layers

### PR DESCRIPTION
**Pull request by polquant**
Some improvements in models

### CODE REVIEW

File: train_config.yml [hunk 1]
```diff
@@ -50,6 +50,9 @@ transformer:
     dropout_rate: 0.2
 
 heads:
+    model:
+        additional_layers: 0
```
Comment (polquant) [inline train_config.yml:54]: value 0 continues behavior before changes, ie. single linear projection layer in classification heads

File: models/model.py [hunk 2]
```diff
@@ -146,7 +153,7 @@ def forward(self, input, **kwargs):
         x = self.transformer(x)
 
         for head in self.cls_head:
-            predictions.append(head(x.cpu()))
+            predictions.append(head(x))
```
Comment (polquant) [inline models/model.py:156]: when we have self.cls_head as a ModuleList we can keep data in gpu until we calculate a loss

File: models/model.py [hunk 3]
```diff
@@ -95,10 +95,17 @@ def __init__(
...
         for value in self.classification_heads.values():
-            self.cls_head.append(nn.Linear(transformer_parameters["output_size"], value["num_class"]))
+            self.cls_head.append(
+                HeadClassificationSequentialModel(
+                    classes_number=value["num_class"],
+                    representation_size=transformer_parameters["output_size"],
+                    additional_layers=heads["model"]["additional_layers"],
+                    dropout_rate=heads["model"]["dropout_rate"]
+                )
+            )
```
Comment (polquant) [inline models/model.py:108]: Option to process output from transformer through a class head dedicated sequential model before projection to the output linear layer.

Comment (polquant) [reply models/model.py:108]: Because in that directory we have other models as well. Maybe you have a good name for that

**Review by mariaogryczak — APPROVED** [review]

**Review by majsylw — APPROVED** [review]
